constellations/gemini-admin/src/emailProgram/emailPrograms.js

```diff
@@ -0,0 +1,230 @@
+"use strict";
+require("./emailPrograms.sass");
+import axios from "axios";
+import React from "react";
+import API, { executeApiCalls } from "../api.js";
+import { getFullName } from "../utils/userUtils.js";
+
+export class EmailProgramsPage extends React.Component {
+    state = {
+        selectProgramId: "",
+        selectedProgramName: "",
+        programs: [],
+        classes: [],
+        classesForProgram: [],
+        userIds: [],
+        usersInClass: [],
+        selectedUsers: [],
+    };
+
+    componentDidMount = () => {
+        const apiCalls = [
+            API.get("api/programs/all"),
+            API.get("api/classes/all"),
+        ];
+        axios
+            .all(apiCalls)
+            .then(
+                axios.spread((...responses) => {
+                    const programs = responses[0].data;
+                    const classes = responses[1].data;
+                    const hasClassId = responses.length > 3;
+                    let classObj = hasClassId ? responses[3].data : {};
+                    let selectedProgramId = hasClassId
+                        ? classObj.programId
+                        : programs[0].programId;
+
+                    this.setState({
+                        selectProgramId: selectedProgramId,
+                        programs: programs,
+                        classes: classes,
+                    });
+                })
+            )
+            .catch((err) => {
+                console.log("Error: api call failed. " + err.message);
+            });
+    };
+
+    handleProgramChange = (event, value) => {
+        var selectedProgram = this.state.programs.find(
+            (program) => program.programId === event.target.value
+        );
+        selectedProgram = selectedProgram.name;
+        let classesForProgram = this.state.classes.filter(
+            (classes) => classes.programId == event.target.value
+        );
+        const classes = classesForProgram.map((classes) => classes.classId);
+
+        this.setState({
+            [value]: event.target.value,
+            classesForProgram: classes,
+            selectedProgramName: selectedProgram,
+            selectedUsers: [],
+            usersInClass: [],
+        });
+    };
+
+    onCheckClass = (e, classId) => {
+        if (e.target.checked) {
+            API.get("api/user-classes/class/" + classId)
+                .then((response) => {
+                    const users = response.data;
+                    const userIds = users.map((user) => user.userId);
+                    userIds.map((userId) => {
```

Comment:
remember map() is used to create a new list.
Use .forEach() if you're pushing things to an existing list.